plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts

```diff
@@ -138,4 +165,15 @@ export class DefaultTechDocsCollator implements DocumentCollator {
   ) {
     return `${techDocsBaseUrl}/static/docs/${entityInfo.namespace}/${entityInfo.kind}/${entityInfo.name}/search/search_index.json`;
   }
+
+  private static handleEntityInfoCasing(
+    legacyPaths: boolean,
+    entityInfo: EntityInfo,
+  ): EntityInfo {
+    return legacyPaths
+      ? entityInfo
+      : Object.entries(entityInfo).reduce((acc, [key, value]) => {
+          return { ...acc, [key]: value.toLowerCase() };
```

Comment:
always `toLocaleLowerCase('en-US')` to be sure that you don't run into e.g. this https://en.wikipedia.org/wiki/Dotted_and_dotless_I